tests/unit/test_loaders.py

```diff
@@ -36,42 +43,80 @@
 class TestJSONFileLoader(BaseEnvVar):
     def setUp(self):
         super().setUp()
-        self.data_path = os.path.join(os.path.dirname(__file__), 'data')
+        self.data_path = pathlib.Path(__file__).parent.joinpath('data')
+        self.zip_data_path = ZipPath(self.data_path.joinpath('Archive.zip'))
         self.file_loader = JSONFileLoader()
-        self.valid_file_path = os.path.join(self.data_path, 'foo')
-        self.compressed_file_path = os.path.join(self.data_path, 'compressed')
+        self.valid_file_path = self.data_path.joinpath('foo')
+        self.zip_valid_file_path = self.zip_data_path.joinpath('foo')
+        self.compressed_file_path = self.data_path.joinpath('compressed')
+        self.zip_compressed_file_path = self.zip_data_path.joinpath(
+            'compressed'
+        )
 
     def test_load_file(self):
         data = self.file_loader.load_file(self.valid_file_path)
         self.assertEqual(len(data), 3)
         self.assertTrue('test_key_1' in data)
 
+        zip_data = self.file_loader.load_file(self.zip_valid_file_path)
+        self.assertEqual(len(zip_data), 3)
+        self.assertTrue('test_key_1' in zip_data)
+
+        self.assertEqual(data, zip_data)
+
```

Comment:
This should be a distinct test so it's clear what's failing, i.e. `test_load_file_zip`